### PR TITLE
Break dependency between core/delivery module

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

The delivery module had an unnecessary dependency on the core module - really this should be the other way round where the core module consumes the delivery module without knowing how its internals works. This is a pre-requisite to implement payload storage using the `IntakeService`.

## Testing

Relied on existing test coverage.

### CODE REVIEW

**Review by bidetofevil — APPROVED** [review]
Rad

File: embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/intake/IntakeService.kt [hunk 1]
```diff
@@ -14,10 +13,15 @@ import io.embrace.android.embracesdk.internal.payload.Envelope
...
 
     /**
      * Stores a payload on disk as its JSON representation.
      */
     fun take(intake: Envelope<*>, metadata: StoredTelemetryMetadata)
+
+    /**
+     * Gracefully shutdown the service.
+     */
+    fun shutdown()
```
[the remark this answers is not in the document]
Comment (fractalwrench) [reply embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/intake/IntakeService.kt]: Added an interface

**Comment by fractalwrench** [comment]
Re-requesting review as I had to create a new module to pass the scheduling service tests, which require Clock/BackgroundWorker. IMO this is the best way to avoid a circular dependency between core/delivery modules.

**Review by bidetofevil — APPROVED** [review]
LGTM

**Pull request merged**